src/webgpu/api/validation/capability_checks/limits/maxVertexBuffers.spec.ts

```diff
@@ -0,0 +1,97 @@
+import { range } from '../../../../../common/util/util.js';
+
+import { kRenderEncoderTypes, kLimitBaseParams, makeLimitTestGroup } from './limit_utils.js';
+
+function getPipelineDescriptor(device: GPUDevice, testValue: number): GPURenderPipelineDescriptor {
+  const code = `
+  struct VSInput {
+    ${range(testValue, i => `@location(${i}) p${i}: f32,`).join('\n')}
```

Comment:
nit: Arguably two different things go past the limit here: the `@location` in the shader and the size of `vertex.buffers` below (though I know you can't have exceed the former without the latter).

IIRC you don't actually need to define a location in the shader in order to create a pipeline that uses it. So I think may as well not, to make sure implementations are checking the correct one? (Or have cases for both?)

Reply:
updated to do both